pandas/core/dtypes/cast.py

```diff
@@ -1654,12 +1654,33 @@ def cast_scalar_to_array(
 
     """
     if dtype is None:
-        dtype, fill_value = infer_dtype_from_scalar(value)
+        dtype, value = infer_dtype_from_scalar(value)
     else:
-        fill_value = value
+        if not isinstance(dtype, np.dtype):
+            dtype = np.dtype(dtype)
```

Comment:
is there no risk of getting here with an EADtype?

Reply:
As it is, no, `cast_scalar_to_array` is called after having checked dtype is not EADtype. It must have been written with that in mind.

Reply:
can you add an assert isinstance(dtype, (type(None), str, np.dtype)) and a comment (that's what the type annotation indicates) (at the top of the function)

Reply:
done